### ISSUE TITLE
Redirect not working when adding custom status to response

### ISSUE THREAD

**Issue by jahvi** [forge-42/remix-toast#34]
When I try returning a `redirectWithError` with a custom `status` I get my toast notification displayed just fine but the actual redirect doesn't happen. If I remove the custom status code it works as expected.

Eg, this works:
```
return redirectWithError("/my-page", "Error");
```

But this doesn't:
```
return redirectWithError("/my-page", "Error", { status: 400 });
```

Do I need to handle merging the headers?

I'm using v1.2.2

**Comment by bravo-kernel** [comment]
Same for `dataWithError`, without the status code the toast functions as expected.

**Comment by bravo-kernel** [comment]
It works when passing a full Response like this.

```
  if (errors) {
    return dataWithError(errors, "Form data rejected by server", new Response(null, {
      status: 422,
    }))
  }
```

@AlemTuzlak the type hint looks incorrect . 

```ts
init?: ResponseInit
```

**Comment by AlemTuzlak** [comment]
@jahvi I'm pretty sure the reason behind it is that you rewrite the redirect status code which is 403 and react router doesn't pick it up as a redirect probably, but dataWithError should work fine, would you mind creating a minimum repro and i'll look at it and see if it's the libraries fault or not. Maybe it's something with the fact that the behaviors changed with rr7 slightly

**Comment by bravo-kernel** [comment]
@AlemTuzlak has this been confirmed/completed via a new release?

**Comment by AlemTuzlak** [comment]
@bravo-kernel no, but no repro either, also certain status codes end up in error boundaries or count as redirects so not sure if it's a bug or just expected behavior
